packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -356,11 +356,8 @@ class _DropdownMenuRouteLayout<T> extends SingleChildLayoutDelegate {
     assert(() {
       final Rect container = Offset.zero & size;
       if (container.intersect(buttonRect) == buttonRect) {
-        // If the button was entirely on-screen, then verify
-        // that the menu is also on-screen.
-        // If the button was a bit off-screen, then, oh well.
+        // Verify button fits on the screen.
```

Comment:
shouldn't the fix to be in the getMenuLimits to not return something that does not fit in the size?